src/Avalonia.Visuals/Media/Fonts/FontFamilyLoader.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Avalonia.Platform;
+
+namespace Avalonia.Media.Fonts
+{
+    internal static class FontFamilyLoader
+    {
+        private static readonly Dictionary<string, AssemblyDescriptor> s_assemblyNameCache
+            = new Dictionary<string, AssemblyDescriptor>();
+
+        private static readonly AssemblyDescriptor s_defaultAssembly;     
+
+        static FontFamilyLoader()
+        {
+            s_defaultAssembly = new AssemblyDescriptor(Assembly.GetEntryAssembly());       
+        }
+
+        public static IEnumerable<FontResource> LoadFontResources(FontFamilyKey fontFamilyKey)
+        {
+            return fontFamilyKey.FileName != null
+                ? GetFontResourcesByFileName(fontFamilyKey.Location, fontFamilyKey.FileName)
+                : GetFontResourcesByLocation(fontFamilyKey.Location);
+        }
+
+        /// <summary>
+        /// Searches for font resources at a given location and returns a quanity of found resources
+        /// </summary>
+        /// <param name="location"></param>
+        /// <returns></returns>
+        private static IEnumerable<FontResource> GetFontResourcesByLocation(Uri location)
+        {
+            var assembly = GetAssembly(location);
```

Comment:
You shouldn't be accessing reflection API directly. That's what IAssetLoader is there for.